### DISCUSSION TITLE
v1.14.0: Medicine stock tracking

### DISCUSSION THREAD

**Discussion by Futsch1** [Futsch1/medTimer#406] [Announcements]
A long awaited and much requested feature has been added in v1.14.0: Tracking of medicine stock. For this, a new icon was added to the medicine view (the box):

<img src='https://github.com/user-attachments/assets/d63dd4bb-c98a-4b32-95f1-3e7c9f50d414' width=300/>

For each medicine, the amount in stock can be set, a potential reminder can be setup and a convenient refill button increases the amount by a configurable pack size. These reminder appear as soon as the stock level has reached the indicated threshold. Medicine stock levels are also shown in standard reminders and in the medicine list.

<img src='https://github.com/user-attachments/assets/184257f2-a44a-4bf4-a26b-fa480b56b813' width=300/>

Significant changes were made under the hood to make automatic testing much more stable and reducing multi-threading complexity.

**Comment by alephalpha0** [comment]
Ok.. so obviously, the stock management screen has had an update since this version... And it's confusing the ever loving hell out of me. 

What is the intended meaning/function of **Amount** and **Unit**?? 
<img width="1080" height="2400" alt="Screenshot_20250927-231639 MedTimer" src="https://github.com/user-attachments/assets/bf4a4322-78c6-4340-91e9-701664c6c585" />
I'm pretty sure I am getting them backwards. Is amount the physical count of how many of the medication is on hand, and unit the amount that is consumed at each dose?

If that's the case, why does my medication card state (9 1 left)??
<img width="1080" height="2400" alt="ResizedImage_Screenshot_20250927-231657 MedTimer2025-09-27_23-18-28_1957" src="https://github.com/user-attachments/assets/b90748a4-c70e-4171-ab85-398010e4a695" />
Or is it actually stating that there are 91 left?

**Reply by alephalpha0** [reply]
🤦
Ok. I finally figured it. Was making it waaaay harder than it actually was. **Unit** is used for the physical description of the medication... Pills, or doses, or whatever arbitrary thing you wanna call it. **Amount** is the stock on hand, pill count or oz of cream.. And dosage for notifications is the amount you subtract from **Stock Amount** each time you take your meds.

**Reply by Futsch1** [reply]
You figured it out... Could you suggest potential improvements how to make it more clear? Both in terms of naming and maybe also in terms of UI? What would have helped you understand it quicker?

**Reply by csglr** [reply]
I had the same issue. I assumed the dosage required an actual quantity (e.g., 2 oz), but you only need to enter the number of tablets (or similar). In my case, the amount was reduced by 2 units (i.e., 2 oz) after each intake instead of by 1 tablet. Now that I’ve figured it out, everything works perfectly! Great app.

**Comment by Regina3000** [comment]
Perhaps changing the word "amount" to "quantity remaining" would be more clear? And maybe changing "unit" to "unit (mg, ml, etc.)" would be more descriptive? Or maybe having some sample text in the unit text box?

And then someome could add instructions to help clarify.
